### ISSUE TITLE
Alt+1 doesn't properly work on High DPI displays when trying to set window scale to 1.00

### ISSUE THREAD

**Issue by Artoriuz** [mpv-player/mpv#7770]
### Important Information

Provide following Information:
- mpv version: latest shinchiro git (2020-05-24) 685bd6a.
- Windows Version: Windows 10 1909.
- Source of the mpv binary: sourceforge.
- If known which version of mpv introduced the problem: Likely after 5676e5ba3998b5059d358ede05ee54c059ed49db.
- Possible screenshot or video of visual glitches: https://0x0.st/ifiy.mkv, https://i.imgur.com/PAy0QGU.mp4

### Reproduction steps

1) Open file with mpv.
2) mpv's main window is created respecting the display's DPI (125% window scale in my case).
3) Press alt+1 to scale it down to 100%.
4) It doesn't work (it stays at 1.25).
5) Press alt+2 to scale it up to 200%.
6) It works.
7) Press alt+1 to scale it down to 100% (now from 200%).
8) It also works (and goes to 1.00 instead of 1.25).

### Expected behavior

Alt+1 should always be able to set the window scale to 1.00.

### Actual behavior

Alt+1 only works after alt+2.

In order words, the window only resizes to a 1.00 if you resize it to 2.00 first pressing alt+2.

### Log file

https://0x0.st/ifiW.txt

### Sample files

Probably not needed, the issue was reproducible with every single file I tried.

**Comment by wm4** [comment]
window-scale=1 is (as of recently) the native scale factor. So it works as intended.

**Comment by Artoriuz** [comment]
Shouldn't it always set it to 1.25 then in my case? 

Staying at 1.25 when pressing for the first time, but then going to 1.00 if you're at 2.00 (after pressing alt+2) sounds like unintended behaviour.

But I mean, this doesn't impact day to day use at all so even if it indeed a bug it's very low priority.

**Comment by wm4** [comment]
Depends if you're talking about the property value, the current-window-scale, or the "actual" scaling.

**Comment by Artoriuz** [comment]
When I open a 1280x690 file it opens as 1600x862 (respecting the 125% DPI scaling factor on Windows) and is displayed as 1.25 window scale on the OSD (pressing i).

Pressing alt+1, from this state, does absolutely nothing and the window stays at 1.25.

Pressing alt+2 changes it to 1.953 (probably because the resulting window would be too big).

Then, after pressing alt+1 again, it actually goes to 1.00 (1280x690) instead of 1.25 (1600x862, where it started).

It just feels weird that it isn't consistent, I think alt+1 should **always** be able to set it to either 1.00 or 1.25, depending on what is the intended behaviour.

I've just noticed it but pressing alt+1 is also broken after you resize the window by dragging any of its edges with the mouse. Instead of scaling it to 1.00 (or 1.25) it simply does nothing.

**Comment by Artoriuz** [comment]
https://0x0.st/ifKa.mkv

Video demonstrating that alt+1 doesn't work after you resize the window down with the mouse.

**Comment by wm4** [comment]
I guess there's something fishy then.

**Comment by garoto** [comment]
@Artoriuz 

See if this .lua bit fix it:

```lua
function fix_window_scale()
    local cws = mp.get_property_native("current-window-scale")
    if cws then mp.set_property_native("window-scale", cws) end
end

mp.register_event("video-reconfig", fix_window_scale)
```

**Comment by Artoriuz** [comment]
@garoto Sorry for the delay. I tested the script and now the window actually goes to 1.00 when alt+1 is pressed. 

The thing is, after resizing with the mouse alt+1 goes back to not doing anything.

**Comment by Artoriuz** [comment]
I retested it with --hidpi-window-scale=no and the problem doesn't seem to have anything to do with high DPI after all.

With --hidpi-window-scale=no, nothing happens if I attempt to put the window back to 100% with alt+1 after resizing it with the mouse dragging its edges.

**Comment by garoto** [comment]
Blame wm4 when he decided, for NO reason whatsoever, to create a new `window-scale` property and while at it, made it non-writtable. Everything was working JUST FINE regarding window-scale manipulation and whatnot, including mouse resizing on Winblows.

But NO, let's brake everyone's use of `scale-window` instead.

**Comment by Dudemanguy** [comment]
Looks like this is fixed by #8247.
